test/irb/test_cmd.rb

```diff
@@ -982,5 +985,20 @@ def test_edit_with_instance_method
       assert_match(/path: .*\/lib\/irb\.rb/, out)
       assert_match("command: ': code'", out)
     end
+
+    def test_edit_with_editor_env_var
+      original_visual = ENV.delete("VISUAL")
+
+      out, err = execute_lines(
+        "edit",
+        irb_path: __FILE__
+      )
+
+      assert_empty err
+      assert_match("path: #{__FILE__}", out)
+      assert_match("command: ': code2'", out)
+
+      ENV["VISUAL"] = original_visual
```

Comment:
Nit: I think recovering `VISUAL` here is not necessary as it'd be done by the teardown callback anyway.

```suggestion
      ENV.delete("VISUAL")

      out, err = execute_lines(
        "edit",
        irb_path: __FILE__
      )

      assert_empty err
      assert_match("path: #{__FILE__}", out)
      assert_match("command: ': code2'", out)
```